diff -Naur a/target/linux/ramips/base-files/etc/board.d/02_network b/target/linux/ramips/base-files/etc/board.d/02_network
--- a/target/linux/ramips/base-files/etc/board.d/02_network	2020-04-30 12:15:30.165523843 +0300
+++ b/target/linux/ramips/base-files/etc/board.d/02_network	2020-04-30 12:32:33.771071855 +0300
@@ -125,6 +125,7 @@
 	sap-g3200u3|\
 	sk-wb8|\
 	telco-electronics,x1|\
+	totolink,a7000r|\
 	totolink,lr1200|\
 	unielec,u7621-06-256m-16m|\
 	unielec,u7621-06-512m-64m|\
diff -Naur a/target/linux/ramips/dts/TOTOLINK-A7000R.dts b/target/linux/ramips/dts/TOTOLINK-A7000R.dts
--- a/target/linux/ramips/dts/TOTOLINK-A7000R.dts     1970-01-01 03:00:00.000000000 +0300
+++ b/target/linux/ramips/dts/TOTOLINK-A7000R.dts     2020-04-30 12:42:33.771071855 +0300
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "totolink,a7000r", "mediatek,mt7621-soc";
+	model = "TOTOLINK A7000R";
+
+	aliases {
+		led-boot = &led_sys;
+		led-failsafe = &led_sys;
+		led-running = &led_sys;
+		led-upgrade = &led_sys;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_sys: sys {
+			label = "a7000r:blue:sys";
+			gpios = <&gpio0 18 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 4 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0 0x30000>;
+				read-only;
+			};
+
+			partition@30000 {
+				label = "u-boot-env";
+				reg = <0x30000 0x10000>;
+				read-only;
+			};
+
+			factory: partition@40000 {
+				label = "factory";
+				reg = <0x40000 0x10000>;
+				read-only;
+			};
+
+			partition@50000 {
+				compatible = "denx,uimage";
+				label = "firmware";
+				reg = <0x50000 0xfb0000>;
+			};
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x0000>;
+		ieee80211-freq-limit = <2400000 2500000>;
+	};
+};
+
+&pcie1 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		mediatek,mtd-eeprom = <&factory 0x8000>;
+		ieee80211-freq-limit = <5000000 6000000>;
+	};
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0xe000>;
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "i2c", "wdt";
+			ralink,function = "gpio";
+		};
+	};
+};
diff -Naur a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
--- a/target/linux/ramips/image/mt7621.mk     2020-04-30 12:15:30.165523843 +0300
+++ b/target/linux/ramips/image/mt7621.mk     2020-04-30 12:32:33.771071855 +0300
@@ -235,6 +235,15 @@
 endef
 TARGET_DEVICES += k2p

+define Device/totolink_a7000r
+	  DTS := TOTOLINK-A7000R
+	  IMAGE_SIZE := 16064k
+	  UIMAGE_NAME := C8340R1C-9999
+	  DEVICE_TITLE := TOTOLINK A7000R
+	  DEVICE_PACKAGES := kmod-mt7615e wpad-basic
+endef
+TARGET_DEVICES += totolink_a7000r
+ 
 define Device/xiaomi_mir3p
 	DTS := MIR3P
 	BLOCKSIZE := 128k
